### PR TITLE
Many small fixes from running REST API tests

### PR DESCRIPTION
**Pull request by sethmlarson**
When switching to code generation for the Python client I found the following issues. These issues apply to 8.x and 7.16, another PR will be raised with 8.x-specific additions.

### CODE REVIEW

File: specification/cat/health/CatHealthRequest.ts [hunk 1]
```diff
@@ -26,7 +26,6 @@ import { CatRequestBase } from '@cat/_types/CatBase'
  */
 export interface Request extends CatRequestBase {
   query_parameters: {
-    include_timestamp?: boolean
```
Comment (sethmlarson) [inline specification/cat/health/CatHealthRequest.ts:29]: Leftover from elasticsearch-net, I went through https://github.com/elastic/elasticsearch-net/search?q=RenameQueryStringParams to find all the remaining issues and fixed them all here.

File: specification/indices/simulate_index_template/IndicesSimulateIndexTemplateRequest.ts [hunk 2]
```diff
@@ -32,11 +40,31 @@ export interface Request extends RequestBase {
...
+     * @server_default 30s
+     */
+    master_timeout?: Time
+  }
   body: {
-    index_patterns?: IndexName[]
+    index_patterns?: Indices
     composed_of?: Name[]
-    /** Any overlapping templates that would have matched, but have lower priority */
-    overlapping?: OverlappingIndexTemplate[]
```
Comment (sethmlarson) [inline specification/indices/simulate_index_template/IndicesSimulateIndexTemplateRequest.ts:39]: `overlapping` isn't documented but is a part of the response. I couldn't find any mention of it in the Elasticsearch repository either. Perhaps this was a mistake and should have been on the response?

File: specification/ml/explain_data_frame_analytics/MlExplainDataFrameAnalyticsRequest.ts [hunk 3]
```diff
@@ -65,7 +65,7 @@ export interface Request extends RequestBase {
      * perform one of the following types of analysis: classification, outlier
      * detection, or regression.
      */
-    analysis: DataframeAnalysisContainer
+    analysis?: DataframeAnalysisContainer
```
Comment (sethmlarson) [inline specification/ml/explain_data_frame_analytics/MlExplainDataFrameAnalyticsRequest.ts:68]: Documented as required, however there is a REST API test that makes a successful request without this field.

File: specification/ml/put_calendar/MlPutCalendarRequest.ts [hunk 4]
```diff
@@ -33,6 +33,10 @@ export interface Request extends RequestBase {
     calendar_id: Id
   }
   body: {
+    /**
+     * An array of anomaly detection job identifiers.
+     */
+    job_ids?: Id[]
```
Comment (sethmlarson) [inline specification/ml/put_calendar/MlPutCalendarRequest.ts:39]: Not documented but used in a REST API test.

File: specification/nodes/hot_threads/NodesHotThreadsRequest.ts [hunk 5]
```diff
@@ -29,14 +29,50 @@ import { Time } from '@_types/Time'
...
+     * returns an error.
+     * @server_default 30s
+     */
+    master_timeout?: Time
+    /**
+     * Specifies the number of hot threads to provide information for.
+     * @server_default 3
+     */
     threads?: long
-    thread_type?: ThreadType
```
Comment (sethmlarson) [inline specification/nodes/hot_threads/NodesHotThreadsRequest.ts:39]: Another leftover from elasticsearch-net.

File: specification/watcher/put_watch/WatcherPutWatchRequest.ts [hunk 6]
```diff
@@ -39,7 +39,7 @@ export interface Request extends RequestBase {
   query_parameters: {
     active?: boolean
     if_primary_term?: long
-    if_sequence_number?: long
+    if_seq_no?: SequenceNumber
```
Comment (sethmlarson) [inline specification/watcher/put_watch/WatcherPutWatchRequest.ts:42]: Another leftover from elasticsearch-net

**Comment by sethmlarson** [comment]
Pinging @lcawl for the machine learning API changes

File: specification/indices/simulate_index_template/IndicesSimulateIndexTemplateRequest.ts (hunk 2 above)
Comment (delvedor) [reply specification/indices/simulate_index_template/IndicesSimulateIndexTemplateRequest.ts:39]: @swallez can you double check this?

**Review by delvedor — APPROVED** [review]

File: specification/indices/simulate_index_template/IndicesSimulateIndexTemplateRequest.ts (hunk 2 above)
Comment (swallez) [reply specification/indices/simulate_index_template/IndicesSimulateIndexTemplateRequest.ts:39]: Confirmed. Request body is at https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java

We can also see an `allow_auto_create` optional boolean there.

**Review by swallez — APPROVED** [review]
LGTM!

File: specification/cluster/health/ClusterHealthRequest.ts [hunk 7]
```diff
@@ -64,6 +64,11 @@ export interface Request extends RequestBase {
      * @server_default 30s
      */
     timeout?: Time
+    /**
+     * A boolean value which controls whether to return HTTP 200 status code instead of HTTP 408 in case of a cluster health timeout from the server side.
+     * @server_default false
+     */
+    return_200_for_cluster_health_timeout?: boolean
```
[the remark this answers is not in the document]
Comment (sethmlarson) [reply specification/cluster/health/ClusterHealthRequest.ts:71]: This option is a 7.16 and 8.0 exclusive. Doesn't look like it's available in 8.1 though?

**Comment by sethmlarson** [comment]
- Removed `return_200_for_cluster_health_timeout` on cluster health
- Added `allow_auto_create` onto simulate index template

**Pull request merged**